### DISCUSSION TITLE
Low fps in high-density (visualization) maps

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#19829] [Q&A] [answered]
In special high-density visualization maps, watching them get played by autoplay is not really possible because autoplay just falls behind the song as it's not able to keep up with the map in real-time (fps drop below 60).
An example of such a map where fps regularly drop below 10 for me is [The Empress](https://osu.ppy.sh/beatmapsets/328780/#osu/729138).
I saw #3252, in which the last statement was that most of the problems brought up in that issue had been resolved, but there were still planned improvements.
I wanted to ask specifically about in-game performance while watching autoplay for such maps: Are there still plans to improve performance in such a scenario? I could imagine any optimizations there to be helpful for normal gameplay as well.

For reference, an example of what is shown as top cpu usage in-game when pressing `Ctrl+F2` (using the default lazer skin):
![image](https://user-images.githubusercontent.com/35152647/185308760-44ea69d8-1395-4c3a-85bd-3d3201f9bba5.png)

**Comment by peppy** [comment, marked as answer]
Performance and optimisation considerations are an ongoing effort. Please be patient and wait for future optimisations.
